src/components/ShopOnMap.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+    <div class="columns">
+        <h3 v-if='location' >User location coords:  {{location[0]}} {{location[1]}}</h3>
+        <gmap-map
+            :center='shopCoords'
+            :zoom='zoom'
+            class='map_shop'
+        >
+            <gmap-marker
+                :key=''
+                :position='shopCoords'
+                :clickable=true
+                :draggable=false
+                @click='shopClicked'
+            >
+            </gmap-marker>
+        </gmap-map>
+        <div></div>
+    </div>
+</template>
+
+<script>
+import * as VueGoogleMaps from 'vue2-google-maps';
+import Vue from 'vue';
+
+Vue.use(VueGoogleMaps, {
+    load: {
+        key: 'AIzaSyB-c3v8sMxEM5nDACJKUFBH4p_Up8eaFUs',
+    },
+});
+
+export default
+{
+    name: 'shopOnMap',
+    methods:
+    {
+        shopClicked ()
+        {
+            // this method will be called to interact with shop
+            // for example to view distance information from the
+            // user
+        },
+    },
+    computed:
+    {
+        location ()
+        {
+            // dislays user location at this moment
+            return this.$store.getters.location;
+        },
+        zoom ()
+        {
+            return this.$store.getters.zoom;
+        },
+        shopCoords ()
+        {
+            return {
+                lat: this.$store.getters.nearbyShops
```

Comment:
I can set it up as I have proposed by setting id in the state and getting it be read by getter than filtered by getter but for Your request We are setting it by router, so access isn't allowed on getters level